server/src/main/java/org/elasticsearch/rest/RestUtils.java

```diff
@@ -146,17 +146,16 @@ private static String decodeComponent(final String s, final Charset charset, boo
     }
 
     private static boolean decodingNeeded(String s, int size, boolean plusAsSpace) {
-        boolean decodingNeeded = false;
-        for (int i = 0; i < size; i++) {
+        if (Strings.isEmpty(s)) {
+            return false;
```

Comment:
This is probably not making anything any faster. An empty string will naturally not execute any iterations of the for loop.

Reply:
You are right. A subconscious boundary condition handling, it is redundant